backend/api/views/quiz.py

```diff
@@ -1,80 +1,77 @@
+from rest_framework.views import APIView
 
-from ..validators import SchemaValidator
 from ..client import client
-from ..utils import JsonResponse, get_sort_order_from_request, get_new_id
 from ..exceptions import InvalidData
+from ..utils import JsonResponse, get_new_id, get_sort_order
+from ..validators import SchemaValidator
+
 
+class QuizApiView(APIView):
+    _quizzes = client.collection("quizzes")
+    _validator = SchemaValidator("quizzes")
 
-class QuizList(APIView):
-    __quizzes = client.collection("quizzes")
-    __validator = SchemaValidator("quizzes")
 
-    def get(self, request: Request) -> JsonResponse:
-        sort_order = get_sort_order_from_request(request, self.__quizzes)
-        return JsonResponse(data=self.__quizzes.find(sort=sort_order))
+class QuizList(QuizApiView):
+    """LIST all quizzes and CREATE new quiz."""
 
-    def post(self, request: Request) -> JsonResponse:
+    def get(self, request: Request):
+        sort_order = get_sort_order(request.query_params, self._quizzes)
+        return JsonResponse(data=self._quizzes.find(sort=sort_order))
+
+    def post(self, request: Request):
         quiz_data = request.data
-        quiz_data["id"] = get_new_id(self.__quizzes)
+        quiz_data["id"] = get_new_id()
 
         try:
-            self.__validator.validate(quiz_data)
+            self._validator.validate(quiz_data)
         except InvalidData as error:
             return JsonResponse(
                 data={"message": f"Invalid JSON: {str(error)}"},
                 status=status.HTTP_400_BAD_REQUEST,
             )
         else:
-            insertion_result = self.__quizzes.insert_one(quiz_data)
-            inserted_quiz = self.__quizzes.find_one(
+            insertion_result = self._quizzes.insert_one(quiz_data)
+            inserted_quiz = self._quizzes.find_one(
                 {"_id": insertion_result.inserted_id}
             )
             return JsonResponse(data=inserted_quiz, status=status.HTTP_201_CREATED)
 
 
-class QuizDetails(APIView):
-    __quizzes = client.collection("quizzes")
-    __validator = SchemaValidator("quizzes")
+class QuizDetail(QuizApiView):
+    """GET, PATCH and DELETE a single quiz."""
 
-    def get(self, request: Request, pk: str) -> JsonResponse:
-        if request.method == "GET":
-            quiz = self.__quizzes.find_one({"id": pk})
+    def get(self, request: Request, pk: str):
+        quiz = self._quizzes.find_one({"id": pk})
 
-            if quiz is None:
-                raise Http404("There is no quiz matching your query.")
-            else:
-                return JsonResponse(data=quiz)
+        if quiz is None:
+            raise Http404("There is no quiz matching your query.")
+        else:
+            return JsonResponse(data=quiz)
 
     def patch(self, request: Request, pk: str) -> JsonResponse:
-        if request.method == "PATCH":
-            quiz_data = request.data
+        quiz_data = request.data
 
-            try:
```

Comment:
I was thinking about implementing an iterator for this `try, except, else` block but it seemed like too much and overengineered.

Reply:
Yeah, way too much...